cmd/cloud.go

```diff
@@ -672,6 +672,221 @@ func updateDeploymentStatusCompleted(logger logger.Logger, apiUrl, token, deploy
+			if err != nil {
+				tui.ShowError("%s", err.Error())
+				os.Exit(1)
+				return
+			}
+			tui.ShowSuccess("Deployment rolled back successfully")
+		}
+	},
+}
+
+// Helper to fetch projects and prompt user to select one. Returns selected project ID or empty string.
+func cloudSelectProject(ctx context.Context, logger logger.Logger, apiUrl, apikey string, prompt string) string {
+	var projects []project.ProjectListData
+	action := func() {
+		var err error
+		projects, err = project.ListProjects(ctx, logger, apiUrl, apikey)
+		if err != nil {
+			errsystem.New(errsystem.ErrApiRequest, err, errsystem.WithContextMessage("Failed to list projects")).ShowErrorAndExit()
+		}
+	}
+	tui.ShowSpinner("fetching projects ...", action)
+	if len(projects) == 0 {
+		fmt.Println()
+		tui.ShowWarning("no projects found")
+		tui.ShowBanner("Create a new project", tui.Text("Use the ")+tui.Command("new")+tui.Text(" command to create a new project"), false)
+		return ""
+	}
+	var options []tui.Option
+	for _, p := range projects {
+		options = append(options, tui.Option{
+			ID:   p.ID,
+			Text: tui.Bold(tui.PadRight(p.Name, 20, " ")) + tui.Muted(p.ID),
+		})
+	}
+	selected := tui.Select(logger, prompt, "", options)
+	if selected == "" {
+		tui.ShowWarning("no project selected")
+	}
+	return selected
+}
+
+func cloudSelectDeployment(ctx context.Context, logger logger.Logger, apiUrl, apikey, projectId string, prompt string) string {
+	var deployments []project.DeploymentListData
+	fetchDeploymentsAction := func() {
+		var err error
+		deployments, err = project.ListDeployments(ctx, logger, apiUrl, apikey, projectId)
+		if err != nil {
+			errsystem.New(errsystem.ErrApiRequest, err, errsystem.WithContextMessage("Failed to list deployments")).ShowErrorAndExit()
+		}
+	}
+	tui.ShowSpinner("fetching deployments ...", fetchDeploymentsAction)
+	if len(deployments) == 0 {
+		tui.ShowWarning("no deployments found")
+		return ""
+	}
+	var deploymentOptions []tui.Option
+	for _, d := range deployments {
+		date, err := time.Parse(time.RFC3339, d.CreatedAt)
+		if err != nil {
+			errsystem.New(errsystem.ErrApiRequest, err, errsystem.WithContextMessage("Failed to parse deployment date")).ShowErrorAndExit()
+		}
+		var msg string
+		if len(d.Message) > 60 {
+			msg = d.Message[:57] + "..."
+		} else {
+			msg = d.Message
+		}
+		tags := strings.Join(d.Tags, ", ")
+		if len(tags) > 50 {
+			tags = tags[:50] + "..."
+		}
+
+		if d.Active {
+			deploymentOptions = append(deploymentOptions, tui.Option{
+				ID:   d.ID,
+				Text: fmt.Sprintf("%s  %s, tags: [%-50s], msg: [%s]", "✅", tui.Title(date.Format("2006-01-02 15:04:05")), tui.Bold(tags), tui.Bold(msg)),
+			})
+		} else {
+			deploymentOptions = append(deploymentOptions, tui.Option{
+				ID:   d.ID,
+				Text: fmt.Sprintf("    %s, tags: [%-50s], msg: [%s]", tui.Title(date.Format("2006-01-02 15:04:05")), tui.Bold(tags), tui.Bold(msg)),
+			})
+		}
```

Comment:
i don't really love the way this looks... i might make some suggestions /changes here

Reply:
Yeah, not sure what the best way is or what info to show of each deployment